DataAccess/DataAccess.Test/PostgreRepositoriesTest.cpp

```diff
@@ -0,0 +1,257 @@
+#include <catch2/catch.hpp>
+
+#include "DataAccess.Postgre/DataBaseOptions.hpp"
+#include "DataAccess.Postgre/PostgreAdapter.hpp"
+#include "DataAccess.Postgre/PostgreRepositories.hpp"
+
+using DataAccess::RegisterRepository;
+using DataAccess::ChannelsRepository;
+using DataAccess::DirectMessageRepository;
+using DataAccess::LoginRepository;
+using DataAccess::MessagesRepository;
+using DataAccess::PostgreAdapter;
+using DataAccess::PostgreTable;
+
+TEST_CASE("PostgreRepositories test", "[dummy]")
+{
+	auto testEmail{ "anotheruser@gmail.com" };
+	auto testLogin{ "anotheruser" };
+	auto testPassHash{ "65e84be33532fb784c48129675f9eff3a682b27168c0ea744b2cf58ee02337c5" };
+
+	auto testTable = std::make_unique<PostgreTable>("users", PostgreAdapter::Instance());
+	auto testInstance = PostgreAdapter::Instance();
+
+	SECTION("Register repository")
+	{
+		SECTION("Register repos constructor")
+		{
+			REQUIRE_NOTHROW(RegisterRepository(PostgreAdapter::Instance()));
+		}
+
+		SECTION("Register repos constructor but reference already used")
+		{
+			REQUIRE_NOTHROW(RegisterRepository(testInstance));
+		}
+
+		RegisterRepository testRegisterRepos(PostgreAdapter::Instance());
+
+		SECTION("Register user")
+		{
+			Network::RegistrationInfo ourUser(testEmail, testLogin, testPassHash);
+			REQUIRE(testRegisterRepos.registerUser(ourUser) == Utility::RegistrationCodes::SUCCESS);
+		}
+		SECTION("Trying to register new user with data of already existing user")
+		{
+			auto newTestLogin{ "newuser" };
+			Network::RegistrationInfo badUserEmail(testEmail, newTestLogin, testPassHash);
+			REQUIRE(testRegisterRepos.registerUser(badUserEmail) == Utility::RegistrationCodes::EMAIL_ALREADY_EXISTS);
```

Comment:
I suppose you could decrease length of _Utility::RegistrationCodes::..._ by _using_